au4/src/processing/internal/processingactionscontroller.cpp

```diff
@@ -23,6 +31,51 @@ void ProcessingActionsController::notifyActionCheckedChanged(const ActionCode& a
     m_actionCheckedChanged.send(actionCode);
 }
 
+void ProcessingActionsController::clipCut()
+{
+    NOT_IMPLEMENTED;
+}
+
+void ProcessingActionsController::clipCopy()
+{
+    NOT_IMPLEMENTED;
+}
+
+void ProcessingActionsController::clipDelete()
+{
+    project::IAudacityProjectPtr project = globalContext()->currentProject();
+    auto selectedTracks = selectionController()->dataSelectedOnTracks();
+    auto selectedStartTime = selectionController()->dataSelectedStartTime();
+    auto selectedEndTime = selectionController()->dataSelectedEndTime();
+    auto tracks = project->processingProject()->trackList();
+    auto selectedClipKey = selectionController()->selectedClip();
+
+    //! TODO AU4: improve for deleting multiple selected clips
+    // remove single clip when selected via header click
+    if (selectedClipKey.index != nidx) {
+        auto clip = project->processingProject()->clipList(selectedClipKey.trackId)[selectedClipKey.index];
```

Comment:
It would be better to add a method like this: Clip clip(const ClipKey& key) const;
(and it is better to use `at` method to access read-only list element)